platform/mds/mds-web/src/main/resources/webapp/js/controllers.js

```diff
@@ -3865,13 +3890,35 @@
                 callback);
         };
 
-        $scope.setAvailableFieldsForDisplay = function() {
+        $scope.subclassCurrent = null;
+
+        $scope.setAvailableFieldsForDisplay = function(scope) {
             var i;
-            $scope.availableFieldsForDisplay = [];
-            for (i = 0; i < $scope.allEntityFields.length; i += 1) {
-                if (!$scope.allEntityFields[i].nonDisplayable) {
-                    $scope.availableFieldsForDisplay.push($scope.allEntityFields[i]);
+            var availableFieldsForDisplay = [];
+            for (i = 0; i < scope.allEntityFields.length; i += 1) {
+                if (!scope.allEntityFields[i].nonDisplayable) {
+                    availableFieldsForDisplay.push(scope.allEntityFields[i]);
+                }
+            }
+            var subclassnameCurrent = scope.$parent.$parent.$parent.subclassCurrent;
+            if(subclassnameCurrent) {
+                var entitiesT = scope.$parent.$parent.$parent.entitiesDerived;
+                var entitySelected = !entitiesT ? null : entitiesT.filter(function(e) {return e.name === subclassnameCurrent;})[0];
+                var fieldsAdded1 = !entitySelected? null : entitySelected.fieldsAdded;
+                var fieldsAdded = fieldsAdded1.slice(6);  // TODO: remove hack
```

Comment:
Excellent question.  This is the reason for "TODO: remove hack".  It's not obvious.

The fieldsAdded are what I want to be in fieldsAdded: just the fields added based on knowing the discriminator.  Somehow fieldsAdded1 has been contaminated with the base Entity fields.  I need to make sure they are not present so the fields don't get duplicated.

I have a few suspicions of where it came from but after some work I have not isolated the true root cause.  I will take care of the root cause and remove the hack before merge.